tensilelite/Tensile/Components/LraTileAssignment.py

```diff
@@ -152,16 +152,22 @@ def LraTileAssignmentCode(self, writer, kernel, tP, tReg, kReg, tmpVgprRes, divi
         isWmma_v1 = writer.states.asmCaps["HasWMMA_V1"]
 
         # get constant parameter
-        tc               = tP["tensorChar"]
-        tile01           = tP["tile01Idx"]
-        waveWidth        = writer.states.kernel["WavefrontSize"]
-        inputPerThread   = kernel["LocalReadVectorWidth"] if not writer.states.inTailLoop else kernel["MIInputPerThread%s"%tc]
+        tc        = tP["tensorChar"]
+        tile01    = tP["tile01Idx"]
+        waveWidth = writer.states.kernel["WavefrontSize"]
+        lrvw      = kernel["LocalReadVectorWidth"]
+
         if kernel["ProblemType"]["Sparse"]:
-          if (kernel["ProblemType"]["Sparse"] == 2 and tP["isB"]) or (kernel["ProblemType"]["Sparse"] == 1 and  tP["isA"]):
-            inputPerThread = inputPerThread // 2
+          if (kernel["ProblemType"]["Sparse"] == 2 and tP["isB"]) or (kernel["ProblemType"]["Sparse"] == 1 and tP["isA"]):
+            lrvw = lrvw // 2
           elif tP["isM"]:
-            inputPerThread = inputPerThread // 8
-        LdsPad           = kernel["LdsPad%s" % tc] if kernel["LdsBlockSizePerPad%s" % tc] == 0 else 0
+            lrvw = lrvw // 8
+
+        miInputPerGroup = int(16 / tP["bpeDS"]) if ((tP["bpeDS"] * kernel["MIInputPerThread%s"%tc]) > 24) else kernel["MIInputPerThread%s"%tc]
```

Comment:
why here is not "> 16"?

Reply:
for future develop F6

Reply:
24 is not power of 2. It is better to describe the reason in python with a comment code.